cmd/root.go

```diff
@@ -11,21 +13,34 @@ import (
 )
 
 var (
-	debug bool
-	quiet bool
+	debug      bool
+	quiet      bool
+	cpuprofile string
 )
 
 func init() {
 	rootCmd.PersistentFlags().BoolVar(&debug, "debug", false, "enable verbose output")
 	rootCmd.PersistentFlags().BoolVarP(&quiet, "quiet", "q", false, "do not output to console; use return code to determine success/failure")
-
+	rootCmd.PersistentFlags().StringVarP(&cpuprofile, "cpuprofile", "p", "", "activate cpu profiling via pprof and write to file")
 	rootCmd.AddCommand(exp.ExpCmd)
 }
 
 var rootCmd = &cobra.Command{
 	Use:          "fogg",
 	Short:        "",
 	SilenceUsage: true,
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
```

Comment:
There a 
```suggestion
	PersistentPreRunE: func(cmd *cobra.Command, args []string) {
```
Might as well use the cmd that does the error handling